### ISSUE TITLE
bug: get empty data from etcd when  apisix init

### ISSUE THREAD

**Issue by gy09535** [apache/apisix#3389]
### Issue description
I  add other apisix instance ,  when it init, I can  not get data from etcd.
But the new config can sync when apisix instance is started,  the etcd is  cluster, I add some logs :
```
2021/01/21 08:39:24 [warn] 30#30: *182 [lua] config_etcd.lua:191: sync_data(): init key: /apisix/upstreams from etcd, res: {"user":"apisix","timeout":30,"password":"xxxxxx","http_host":["http:\/\/xxxxxxxx:2379","http:\/\/xxxxxxx:2379","http:\/\/xxxxxxx:2379"],"protocol":"v3","api_prefix":"\/v3","ssl_verify":false,"tls":{"verify":false},"ttl":-1} etcd config: {"user":"apisix","timeout":30,"password":"xxxx","http_host":["http:\/\/xxxxx:2379","http:\/\/xxxxx:2379","http:\/\/xxxxx:2379"],"protocol":"v3","api_prefix":"\/v3","ssl_verify":false,"tls":{"verify":false},"ttl":-1}, context: ngx.timer
2021/01/21 08:39:24 [warn] 30#30: *182 [lua] config_etcd.lua:294: sync_data(): load key: /apisix/upstreams data: {}, context: ngx.timer
```

But I can  get  data use "etcdctl"
```
 env ETCDCTL_API=3 etcdctl --endpoints http://xxxx:2379 --user=apisix:xxxxxxxx  get --from-key /apisix/upstreams

{"id":"334312499069321769","create_time":1608795377,"update_time":1609140334,"nodes":[{"host":"xxxxx","port":80,"weight":1}],"timeout":{"connect":6000,"read":6000,"send":6000},"type":"roundrobin","name":"backend-service"}
/apisix/upstreams/334338203324514857
{"id":"334338203324514857","create_time":1608810698,"update_time":1609229283,"nodes":[{"host":"dashboard.sqb.svc.cluster.local","port":80,"weight":1}],"timeout":{"connect":10000,"read":10000,"send":10000},"type":"roundrobin","name":"xxxxxxx"}
```
log location is 

```
local function sync_data(self)
    if not self.key then
        return nil, "missing 'key' arguments"
    end

    if self.need_reload then
        local res, err = readdir(self.etcd_cli, self.key)
        if not res then
            return false, err
        end

        local dir_res, headers = res.body.node or {}, res.headers
        log.warn("init key: ", self.key, " from etcd, res: ",
                  json.delay_encode(dir_res)," etcd config: ",json.delay_encode(self.etcd_conf))
        if not dir_res then
            return false, err
        end

        if self.values then
            for i, val in ipairs(self.values) do
                if val and val.clean_handlers then
                    for _, clean_handler in ipairs(val.clean_handlers) do
                        clean_handler(val)
                    end
                    val.clean_handlers = nil
                end
            end

            self.values = nil
            self.values_hash = nil
        end

        local changed = false

        if self.single_item then
            self.values = new_tab(1, 0)
            self.values_hash = new_tab(0, 1)

            local item = dir_res
            local data_valid = item.value ~= nil

            if data_valid and self.item_schema then
                data_valid, err = check_schema(self.item_schema, item.value)
                if not data_valid then
                    log.error("failed to check item data of [", self.key,
                              "] err:", err, " ,val: ", json.encode(item.value))
                end
            end

            if data_valid then
                changed = true
                insert_tab(self.values, item)
                self.values_hash[self.key] = #self.values

                item.clean_handlers = {}

                if self.filter then
                    self.filter(item)
                end
            end

            self:upgrade_version(item.modifiedIndex)

        else
            if not dir_res.nodes then
                dir_res.nodes = {}
            end

            self.values = new_tab(#dir_res.nodes, 0)
            self.values_hash = new_tab(0, #dir_res.nodes)

            for _, item in ipairs(dir_res.nodes) do
                local key = short_key(self, item.key)
                local data_valid = true
                if type(item.value) ~= "table" then
                    data_valid = false
                    log.error("invalid item data of [", self.key .. "/" .. key,
                              "], val: ", item.value,
                              ", it shoud be a object")
                end

                if data_valid and self.item_schema then
                    data_valid, err = check_schema(self.item_schema, item.value)
                    if not data_valid then
                        log.error("failed to check item data of [", self.key,
                                  "] err:", err, " ,val: ", json.encode(item.value))
                    end
                end

                if data_valid then
                    changed = true
                    insert_tab(self.values, item)
                    self.values_hash[key] = #self.values

                    item.value.id = key
                    item.clean_handlers = {}

                    if self.filter then
                        self.filter(item)
                    end
                end

                self:upgrade_version(item.modifiedIndex)
            end
        end

        if headers then
            self:upgrade_version(headers["X-Etcd-Index"])
        end

        if changed then
            self.conf_version = self.conf_version + 1
        end

        self.need_reload = false
        log.warn("load key: ", self.key, " data: ", json.delay_encode(self.values))
        return true
    end
```
### Environment

* apisix version (cmd: `apisix version`):2.0
* OS: (cmd: `uname -a`)
* OpenResty / Nginx version: (cmd: `nginx -V` or `openresty -V`)

### Minimal test code / Steps to reproduce the issue

1.
2.
3.

**Comment by Firstsawyou** [comment]
This problem may have been fixed, you can try the latest version 2.2 .

**Comment by gy09535** [comment]
> This problem may have been fixed, you can try the latest version 2.2 .

which pr fix this ?

**Comment by spacewander** [comment]
@gy09535 
Why the output of `env ETCDCTL_API=3 etcdctl --endpoints http://xxxx:2379 --user=apisix:xxxxxxxx  get --from-key /apisix/upstreams` doesn't have:
```
/apisix/upstreams/
init_dir
```

And what's your etcd's version?

**Comment by gy09535** [comment]
> @gy09535
> Why the output of `env ETCDCTL_API=3 etcdctl --endpoints http://xxxx:2379 --user=apisix:xxxxxxxx get --from-key /apisix/upstreams` doesn't have:
> 
> ```
> /apisix/upstreams/
> init_dir
> ```
> 
> And what's your etcd's version?

3.4.13

**Comment by gy09535** [comment]
> @gy09535
> Why the output of `env ETCDCTL_API=3 etcdctl --endpoints http://xxxx:2379 --user=apisix:xxxxxxxx get --from-key /apisix/upstreams` doesn't have:
> 
> ```
> /apisix/upstreams/
> init_dir
> ```
> 
> And what's your etcd's version?

I use  apisix dashboard 2.0 to add the data

**Comment by gy09535** [comment]
> @gy09535
> Why the output of `env ETCDCTL_API=3 etcdctl --endpoints http://xxxx:2379 --user=apisix:xxxxxxxx get --from-key /apisix/upstreams` doesn't have:
> 
> ```
> /apisix/upstreams/
> init_dir
> ```
> 
> And what's your etcd's version?

Key problem is  I can not get data ,the response is empty.

**Comment by spacewander** [comment]
May be fixed by https://github.com/apache/apisix/pull/2690

**Comment by gy09535** [comment]
> May be fixed by #2690

yes, It is the problem , the  path is not set, so it can not read the children data.

**Comment by spacewander** [comment]
Consider solved
